### ISSUE TITLE
[Bug] zos_copy on an existing sequential file which contains special characters failed

### ISSUE THREAD

**Issue by vlodeibm** [ansible-collections/ibm_zos_core#600]
### Bug description

First step : I launch a zos_copy with in src, a text file which contains special characters, the src file on USS partition, the dest file a SEQ file. I use encoding from utf8 to IBM-1047. My dest file does not exist. The result is ok. My dest file is created and the content is ok..
Second step : I launch again this playbook, with the same parameters. That means that in this second test, my dest file exists and contains special characters. The copy failed with the message UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 1: invalid continuation byte\r\n",

The same test, without special characters is ok. 

The file :
![image](https://user-images.githubusercontent.com/122458609/212043366-90dfbb7f-a881-4130-be11-294463bc51e2.png)

### Playbook verbosity output

TASK [ibm.ibm_zos_plum_deploy.zos_deploy : [deploy_modules>add>sequential_copy]: Copy /tmp/plum_deploy/jenkins/cvb1/ansible.oakm_u0l/seq/seqfile1.seqb to NAZARE.WDEPLOY.BNPP.SEQVL.SEQFILE1] ***
task path: /home/vlode/.ansible/collections/ansible_collections/ibm/ibm_zos_plum_deploy/roles/zos_deploy/tasks/task_zos_copy.yml:18
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: JENKINS
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="JENKINS"' -o ConnectTimeout=10 -o 'ControlPath="/home/vlode/.ansible/cp/7bf7af4626"' 9.30.222.103 '/bin/sh -c '"'"'echo ~JENKINS && sleep 0'"'"''
<9.30.222.103> (0, b'/u/jenkins\n', b'')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: JENKINS
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="JENKINS"' -o ConnectTimeout=10 -o 'ControlPath="/home/vlode/.ansible/cp/7bf7af4626"' 9.30.222.103 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /u/jenkins/.ansible/tmp `"&& mkdir "` echo /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941 `" && echo ansible-tmp-1673519028.2810628-8638-200788736874941="` echo /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941 `" ) && sleep 0'"'"''
<9.30.222.103> (0, b'ansible-tmp-1673519028.2810628-8638-200788736874941=/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941\n', b'')
Using module file /home/vlode/.ansible/collections/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py
<9.30.222.103> PUT /home/vlode/.ansible/tmp/ansible-local-83367qvr8dqq/tmp93dvg96e TO /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py
<9.30.222.103> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="JENKINS"' -o ConnectTimeout=10 -o 'ControlPath="/home/vlode/.ansible/cp/7bf7af4626"' '[9.30.222.103]'
<9.30.222.103> (0, b'sftp> put /home/vlode/.ansible/tmp/ansible-local-83367qvr8dqq/tmp93dvg96e /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py\n', b'')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: JENKINS
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="JENKINS"' -o ConnectTimeout=10 -o 'ControlPath="/home/vlode/.ansible/cp/7bf7af4626"' 9.30.222.103 '/bin/sh -c '"'"'chmod u+x /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/ /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py && sleep 0'"'"''
<9.30.222.103> (0, b'', b'')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: JENKINS
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="JENKINS"' -o ConnectTimeout=10 -o 'ControlPath="/home/vlode/.ansible/cp/7bf7af4626"' -tt 9.30.222.103 '/bin/sh -c '"'"'_BPXK_AUTOCVT=ON ZOAU_HOME=/var/usr/lpp/IBM/zoautil LIBPATH=/var/usr/lpp/IBM/zoautil/lib:/var/usr/lpp/IBM/cyp/v3r10/pyz/lib:/lib:/usr/lib:. PATH=/var/usr/lpp/IBM/zoautil/bin:/var/usr/lpp/IBM/cyp/v3r10/pyz/bin:/bin:/var/bin _CEE_RUNOPTS='"'"'"'"'"'"'"'"'FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)'"'"'"'"'"'"'"'"' _TAG_REDIR_ERR=txt _TAG_REDIR_IN=txt _TAG_REDIR_OUT=txt LANG=C /var/usr/lpp/IBM/cyp/v3r10/pyz/bin/python3 /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py && sleep 0'"'"''
<9.30.222.103> (1, b'Traceback (most recent call last):\r\n  File "/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py", line 100, in <module>\r\n    _ansiballz_main()\r\n  File "/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py", line 92, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File "/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py", line 40, in invoke_module\r\n    runpy.run_module(mod_name=\'ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_copy\', init_globals=dict(_module_fqn=\'ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_copy\', _modlib_path=modlib_path),\r\n  File "/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py", line 224, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File "/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py", line 96, in _run_module_code\r\n    _run_code(code, mod_globals, init_globals,\r\n  File "/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py", line 86, in _run_code\r\n    exec(code, run_globals)\r\n  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 2552, in <module>\r\n  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 2499, in main\r\n  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 2179, in run_module\r\n  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 1760, in does_destination_allow_copy\r\n  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/data_set.py", line 590, in is_empty\r\n  File "/u/jenkins/.local/lib/python3.10/site-packages/zoautil_py/datasets.py", line 70, in read\r\n    response = _read(dataset, *args, **kwargs)\r\n  File "/u/jenkins/.local/lib/python3.10/site-packages/zoautil_py/datasets.py", line 40, in _read\r\n    response = call_zoau_library("dtail", options)\r\nUnicodeDecodeError: \'utf-8\' codec can\'t decode byte 0xe9 in position 1: invalid continuation byte\r\n', b'Shared connection to 9.30.222.103 closed.\r\n')
<9.30.222.103> Failed to connect to the host via ssh: Shared connection to 9.30.222.103 closed.
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: JENKINS
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="JENKINS"' -o ConnectTimeout=10 -o 'ControlPath="/home/vlode/.ansible/cp/7bf7af4626"' 9.30.222.103 '/bin/sh -c '"'"'rm -f -r /u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/ > /dev/null 2>&1 && sleep 0'"'"''
<9.30.222.103> (0, b'', b'')
fatal: [cvb1]: FAILED! => {
    "changed": false,
    "failed_when_result": true,
    "invocation": {
        "module_args": {
            "backup": false,
            "dest": "NAZARE.WDEPLOY.BNPP.SEQVL.SEQFILE1",
            "dest_data_set": {
                "record_format": "FB",
                "record_length": 15,
                "space_primary": 2,
                "space_secondary": 1,
                "space_type": "TRK",
                "type": "SEQ"
            },
            "encoding": {
                "from": "UTF-8",
                "to": "IBM-1047"
            },
            "force": true,
            "is_binary": false,
            "remote_src": true,
            "src": "/tmp/plum_deploy/jenkins/cvb1/ansible.oakm_u0l/seq/seqfile1.seqb"
        }
    },
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1,
    "stderr": "Shared connection to 9.30.222.103 closed.\r\n",
    "stderr_lines": null,
    "stdout": "Traceback (most recent call last):\r\n  File \"/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py\", line 100, in <module>\r\n    _ansiballz_main()\r\n  File \"/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py\", line 92, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/u/jenkins/.ansible/tmp/ansible-tmp-1673519028.2810628-8638-200788736874941/AnsiballZ_zos_copy.py\", line 40, in invoke_module\r\n    runpy.run_module(mod_name='ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_copy', init_globals=dict(_module_fqn='ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_copy', _modlib_path=modlib_path),\r\n  File \"/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py\", line 224, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py\", line 96, in _run_module_code\r\n    _run_code(code, mod_globals, init_globals,\r\n  File \"/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py\", line 86, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py\", line 2552, in <module>\r\n  File \"/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py\", line 2499, in main\r\n  File \"/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py\", line 2179, in run_module\r\n  File \"/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py\", line 1760, in does_destination_allow_copy\r\n  File \"/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_uakdjqq6/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/data_set.py\", line 590, in is_empty\r\n  File \"/u/jenkins/.local/lib/python3.10/site-packages/zoautil_py/datasets.py\", line 70, in read\r\n    response = _read(dataset, *args, **kwargs)\r\n  File \"/u/jenkins/.local/lib/python3.10/site-packages/zoautil_py/datasets.py\", line 40, in _read\r\n    response = call_zoau_library(\"dtail\", options)\r\nUnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 1: invalid continuation byte\r\n",
    "stdout_lines": null
}

### Contents of `ansible.cfg`

```YAML
#*******************************************************************************
# Licensed Materials - Property of IBM
# (c) Copyright IBM Corp. 2022. All Rights Reserved.
#
# Note to U.S. Government Users Restricted Rights:
# Use, duplication or disclosure restricted by GSA ADP Schedule
# Contract with IBM Corp.
#*******************************************************************************
[defaults]
forks = 25
host_key_checking = False
#callbacks_enabled = timer, profile_tasks, profile_roles
stdout_callback = ibm.ibm_zos_plum_deploy.cb_evidences

[ssh_connection]
pipelining = True
#ssh_args = -o ControlMaster=auto -o ControlPersist=3600s -o PreferredAuthentications=publickey

[callback_evidences]
evidences_folder = ../evidences
```

### Contents of the inventory

```YAML
all:
  hosts:
    cvb1:
      ansible_host: X.X.X.X
      ansible_user: MYUSER
      ansible_ssh_port: 222
```

### Contents of `group_vars` or `host_vars`

```YAML
environment_vars:
  _BPXK_AUTOCVT: "ON"
  ZOAU_HOME: "{{ ZOAU }}"
  #PYTHONPATH: "{{ ZOAU }}/lib"
  LIBPATH: "{{ ZOAU }}/lib:{{ PYZ }}/lib:/lib:/usr/lib:."
  PATH: "{{ ZOAU }}/bin:{{ PYZ }}/bin:/bin:/var/bin"
  _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
  _TAG_REDIR_ERR: "txt"
  _TAG_REDIR_IN: "txt"
  _TAG_REDIR_OUT: "txt"
  LANG: "C"
```

### Ansible version

```shell
ansible [core 2.11.12]
  config file = None
  configured module search path = ['/home/vlode/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/vlode/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Jun 22 2022, 20:18:18) [GCC 9.4.0]
  jinja version = 2.10.1
  libyaml = True
```

### IBM z/OS Ansible core Version

v1.4.0

### IBM ZOAU version

v1.2.1

### z/OS version

2.4

### Ansible module

zos_copy

**Comment by fernandofloresg** [comment]
I was not able to reproduce the same, with playbook : 

```
- hosts : zvm
  collections : 
    - ibm.ibm_zos_core
  vars:
    ZOAU: "/zoau/v1.2.1"
    PYZ: "/python2/usr/lpp/IBM/cyp/v3r2/pyz"
    SOURCE_DATASET: "OMVSADM.COPYTEST.SOURCE"
    TARGET_DATASET: "OMVSADM.COPYTEST.DEST"
  environment: "{{ environment_vars }}"
  tasks:
    - name: remove dataset
      zos_data_set:
        name: OMVSADM.SPECIALC.TEST
        state: absent
    
    # first we copy the file to the remote
    - name: Copy file with special chars to EC
      zos_copy:
        src: './files/special_chars'
        dest: '/tmp/special_chars'
        force: True

    # then we copy the remote USS file to a SEQ dataset
    - name: copy uss with special chars to SEQ
      zos_copy:
        src: "/tmp/special_chars"
        remote_src: True
        force: True
        dest: OMVSADM.SPECIALC.TEST
        backup: False
        dest_data_set:
          record_format: "FB"
          record_length: 15
          space_primary: 2
          space_secondary: 1
          space_type: "TRK"
          type: "SEQ"
        encoding:
          from: "UTF-8"
          to: "IBM-1047"
    
    - name: copy uss with special chars to SEQ
      zos_copy:
        src: "/tmp/special_chars"
        remote_src: True
        force: True
        dest: OMVSADM.SPECIALC.TEST
        backup: False
        dest_data_set:
          record_format: "FB"
          record_length: 15
          space_primary: 2
          space_secondary: 1
          space_type: "TRK"
          type: "SEQ"
        encoding:
          from: "UTF-8"
          to: "IBM-1047"
```


Kindly provide which special characters were used and the playbook used?

**Comment by vlodeibm** [comment]
As answered in slack, here is my playbook: #*******************************************************************************
# Licensed Materials - Property of IBM
# (c) Copyright IBM Corp. 2023. All Rights Reserved.
#
# Note to U.S. Government Users Restricted Rights:
# Use, duplication or disclosure restricted by GSA ADP Schedule
# Contract with IBM Corp.
#*******************************************************************************
---
- name: "[{{current_step_name}}]"
  collections:
    - ibm.ibm_zos_core
  environment: "{{ environment_vars }}"
​
  block:
    - name: "[{{current_step_name}}]: Copy without special characters"
      zos_copy:
       src: "/tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqA.seqa"
       dest: "NAZARE.WDEPLOY.BNPP.SEQTSTA"
       remote_src: true
       encoding:
         from: UTF-8
         to: IBM-1047
       backup: false
       force: true
​
    - name: "[{{current_step_name}}]: Copy with special characters"
      zos_copy:
       src: "/tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqfile1.seqb"
       dest: "NAZARE.WDEPLOY.BNPP.SEQTSTB"
       remote_src: true
       encoding:
         from: UTF-8
         to: IBM-1047
       backup: false
       force: true
  tags:
    - sequential_copy

And the content of my initial file with special characters
 &é"'(-è_çà)=FIN
~#{[|`\^@]}FIN
^$ù*,;:!¤FIN
¨£%µ?./§FIN

The first launch of the playbook is ok, the second one failed

Here the result of the failed one :

ansible_host: 9.30.222.103
deploy_date: '20230113_093103'
inventory_hostname: cvb1
status: Failed
activities:
- name: SEQUENTIAL_COPY_TEST
  actions:
  - name: SEQUENTIAL_COPY_TEST
    steps:
    - name: SEQUENTIAL_COPY_TEST
      step_result:
        status: Failed
        msg: 'sequential_copy_test.yml : Failed'
        results:
        - status: Ok
          msg: 'sequential_copy_test.yml line 16 : ibm.ibm_zos_plum_deploy.zos_deploy
            : [sequential_copy_test>sequential_copy_test>sequential_copy_test]: Copy
            without special characters'
          results:
            register: None
            args:
              backup: false
              dest: NAZARE.WDEPLOY.BNPP.SEQTSTA
              encoding:
                from: UTF-8
                to: IBM-1047
              force: true
              remote_src: true
              src: /tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqA.seqa
            result:
              dest: NAZARE.WDEPLOY.BNPP.SEQTSTA
              invocation:
                module_args:
                  backup: false
                  dest: NAZARE.WDEPLOY.BNPP.SEQTSTA
                  encoding:
                    from: UTF-8
                    to: IBM-1047
                  force: true
                  remote_src: true
                  src: /tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqA.seqa
              is_binary: false
              src: /SYSTEM/tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqA.seqa
        - status: Failed
          msg: 'sequential_copy_test.yml line 27 : ibm.ibm_zos_plum_deploy.zos_deploy
            : [sequential_copy_test>sequential_copy_test>sequential_copy_test]: Copy
            with special characters'
          results:
            args:
              backup: false
              dest: NAZARE.WDEPLOY.BNPP.SEQTSTB
              encoding:
                from: UTF-8
                to: IBM-1047
              force: true
              remote_src: true
              src: /tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqfile1.seqb
            result:
              invocation:
                module_args:
                  backup: false
                  dest: NAZARE.WDEPLOY.BNPP.SEQTSTB
                  encoding:
                    from: UTF-8
                    to: IBM-1047
                  force: true
                  remote_src: true
                  src: /tmp/plum_deploy/jenkins/cvb1/ansible.6ewuk88z/seq/seqfile1.seqb
              msg: |-
                MODULE FAILURE
                See stdout/stderr for the exact error
              rc: 1
              stderr: Shared connection to 9.30.222.103 closed.
              stdout: |-
                Traceback (most recent call last):
                  File "/u/jenkins/.ansible/tmp/ansible-tmp-1673598679.2069974-2058-118310920748063/AnsiballZ_zos_copy.py", line 100, in <module>
                    _ansiballz_main()
                  File "/u/jenkins/.ansible/tmp/ansible-tmp-1673598679.2069974-2058-118310920748063/AnsiballZ_zos_copy.py", line 92, in _ansiballz_main
                    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
                  File "/u/jenkins/.ansible/tmp/ansible-tmp-1673598679.2069974-2058-118310920748063/AnsiballZ_zos_copy.py", line 40, in invoke_module
                    runpy.run_module(mod_name='ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_copy', init_globals=dict(_module_fqn='ansible_collections.ibm.ibm_zos_core.plugins.modules.zos_copy', _modlib_path=modlib_path),
                  File "/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py", line 224, in run_module
                    return _run_module_code(code, init_globals, run_name, mod_spec)
                  File "/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py", line 96, in _run_module_code
                    _run_code(code, mod_globals, init_globals,
                  File "/var/usr/lpp/IBM/cyp/v3r10/pyz/lib/python3.10/runpy.py", line 86, in _run_code
                    exec(code, run_globals)
                  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_krbwj2yn/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 2552, in <module>
                  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_krbwj2yn/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 2499, in main
                  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_krbwj2yn/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 2179, in run_module
                  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_krbwj2yn/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py", line 1760, in does_destination_allow_copy
                  File "/tmp/ansible_ibm.ibm_zos_core.zos_copy_payload_krbwj2yn/ansible_ibm.ibm_zos_core.zos_copy_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/data_set.py", line 590, in is_empty
                  File "/u/jenkins/.local/lib/python3.10/site-packages/zoautil_py/datasets.py", line 70, in read
                    response = _read(dataset, *args, **kwargs)
                  File "/u/jenkins/.local/lib/python3.10/site-packages/zoautil_py/datasets.py", line 40, in _read
                    response = call_zoau_library("dtail", options)
                UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 1: invalid continuation byte

**Comment by vlodeibm** [comment]
@ddimatos Hi, what about this issue. I can see "Status in plan" but the label "waiting for response". I gave an answer last week. Do you need more informations? In France, special characters are very used. And for the moment, I did not found any workaround to avoid this issue. So, I can consider this issue as a major issue :(

**Comment by ddimatos** [comment]
@vlodeibm - I do apologize for any delayed response, the team is currently (me included) inundated with a number of entitled support issues as well as [active in plan work items](https://github.com/orgs/ansible-collections/projects/10/views/29). Our community support comes at a slight cost to our users and that cost is time. We try to address new community raised issues as soon as possible such as this one with the idea with new issues we will triage a recreate and depending on the complexity try to get it in the current plan or put it in plan accordingly. 

In this case it was fortunate it looks as if we were able to fit it which you could follow along as it was put in plan (see screen shot).
<img width="379" alt="image" src="https://user-images.githubusercontent.com/25803172/214716583-eac2f743-0e53-4ea4-b5f7-5db2ba3a5c3c.png">

**Comment by ddimatos** [comment]
@vlodeibm - I am not sure if @rexemin who is assigned to this has started but I would like to ask that you [review the requirements](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/release_notes.html#reference) for our collection for `ibm_zos_core` version 1.4.0 listed below as well:
- Supported by [z/OS V2R3](https://www.ibm.com/support/knowledgecenter/SSLTBW_2.3.0/com.ibm.zos.v2r3/en/homepage.html) or later
- Supported by the [z/OS® shell](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.4.0/com.ibm.zos.v2r4.bpxa400/part1.htm)
- Supported by [IBM Open Enterprise SDK for Python](https://www.ibm.com/products/open-enterprise-python-zos) [3.9](https://www.ibm.com/docs/en/python-zos/3.9)
- Supported by IBM [Z Open Automation Utilities 1.1.0](https://www.ibm.com/docs/en/zoau/1.1.0) and [Z Open Automation Utilities 1.1.1](https://www.ibm.com/docs/en/zoau/1.1.1)

Python 3.10 which appears in your verbosity output indicates you are using Python 3.10 `/var/usr/lpp/IBM/cyp/v3r10/pyz` , **only ZOAU 1.2.0.1 or later works with Python 3.10** and **neither of these are supported** with `ibm_zos_core` v 1.4.0. Please retry with a supported combination. 

The issue you have shared has been a known issue to our team when using unsupported dependency combinations ` UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 1: invalid continuation byte` and more apparent when our collection is used with Python 3.10 with `pipelining=true`. You could try to set your `pipelining=False` under `[connection]` , note that the ini section [has been renamed](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#ansible-pipelining) to **[connection]** your version of Ansible 2.11 ( I have not confirmed there is an alias to `ssh_connection` yet).  

Although not a  real Python issue, the UnicodeDecodeError appeared in  Python 3.10 or later for various reasons, you can [track it here](https://github.com/ansible-collections/ibm_zos_core/issues/570) and seems to be related to your issue. 

Please change your dependencies to a supported combination and retry.

**Comment by jbyibm** [comment]
Hello @ddimatos 
I did some more investigations. And we have the same result with supported versions of ZOAU. But by looking deeper in the log the issue for me is in ZOAU not in zos_copy.

**Comment by jbyibm** [comment]
@ddimatos This is definitely a ZOAU issue. You can this issue.
Sorry.

**Comment by rexemin** [comment]
So after a lot of discussion we decided to change the command we use to check whether a sequential dataset is empty, as a way to mitigate this issue on our end.

**Comment by ddimatos** [comment]
Just some comments:
- this got in earlier than I thought is targeted to 1.5.0-beta.1
- we will need to get this into 1.6.0-beta.1 and when we do, can we look at using [member_exists](https://github.com/ansible-collections/ibm_zos_core/blob/ded7a01c46a9a79a7c90e4a41b2a9e8efa142428/plugins/module_utils/data_set.py#L403)()... instead for the sake of reusing code. 
- Lets think about if this needs to be a 1.4.2 fix, no 1.4.2 is planned while 1.4.1 is not ready yet either, so lets discuss this and see what we want to do.

**Comment by ddimatos** [comment]
Needs to be forward ported to `dev` (which will be 1.6.0-beta.1)
